### DISCUSSION TITLE
Exclude previews from borg backup

### DISCUSSION THREAD

**Discussion by daschmidt1994** [nextcloud/all-in-one#5015] [Questions] [answered]
I have a nextcloud aio running and love it.
It also run the app memories with the preview app. Now my backups are very big. 2 user with 200GB of data. The backup is ober 1Tb, now I think it backup the Previews so. I’m right? If yes, is there a wayto exclude this in the nextcloud aio backup strategy?

**Comment by szaimen** [comment]
Hi, can you post the output of `sudo docker inspect nextcloud-aio-mastercontainer` here?

**Reply by daschmidt1994** [reply]
root@openmediavault:~# sudo docker inspect nextcloud-aio-mastercontainer
[
    {
        "Id": "ef49cdd05fcf4b8f1cd7871ac01fe1a7bfd7d154cba56cae57721bcabff0511f",
        "Created": "2024-07-20T06:51:14.575926328Z",
        "Path": "/start.sh",
        "Args": [],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 2142,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2024-07-21T18:16:47.59288204Z",
            "FinishedAt": "2024-07-21T18:15:03.936314864Z",
            "Health": {
                "Status": "healthy",
                "FailingStreak": 0,
                "Log": [
                    {
                        "Start": "2024-07-22T10:30:21.701604635+02:00",
                        "End": "2024-07-22T10:30:21.783809843+02:00",
                        "ExitCode": 0,
                        "Output": "Connection to 127.0.0.1 80 port [tcp/http] succeeded!\nConnection to 127.0.0.1 8000 port [tcp/*] succeeded!\nConnection to 127.0.0.1 8080 port [tcp/http-alt] succeeded!\nConnection to 127.0.0.1 8443 port [tcp/*] succeeded!\nConnection to 127.0.0.1 9000 port [tcp/*] succeeded!\nConnection to 127.0.0.1 9876 port [tcp/*] succeeded!\n"
                    },
                    {
                        "Start": "2024-07-22T10:30:51.78575681+02:00",
                        "End": "2024-07-22T10:30:51.861390822+02:00",
                        "ExitCode": 0,
                        "Output": "Connection to 127.0.0.1 80 port [tcp/http] succeeded!\nConnection to 127.0.0.1 8000 port [tcp/*] succeeded!\nConnection to 127.0.0.1 8080 port [tcp/http-alt] succeeded!\nConnection to 127.0.0.1 8443 port [tcp/*] succeeded!\nConnection to 127.0.0.1 9000 port [tcp/*] succeeded!\nConnection to 127.0.0.1 9876 port [tcp/*] succeeded!\n"
                    },
                    {
                        "Start": "2024-07-22T10:31:21.862905605+02:00",
                        "End": "2024-07-22T10:31:21.93547696+02:00",
                        "ExitCode": 0,
                        "Output": "Connection to 127.0.0.1 80 port [tcp/http] succeeded!\nConnection to 127.0.0.1 8000 port [tcp/*] succeeded!\nConnection to 127.0.0.1 8080 port [tcp/http-alt] succeeded!\nConnection to 127.0.0.1 8443 port [tcp/*] succeeded!\nConnection to 127.0.0.1 9000 port [tcp/*] succeeded!\nConnection to 127.0.0.1 9876 port [tcp/*] succeeded!\n"
                    },
                    {
                        "Start": "2024-07-22T10:31:51.937322827+02:00",
                        "End": "2024-07-22T10:31:52.009084498+02:00",
                        "ExitCode": 0,
                        "Output": "Connection to 127.0.0.1 80 port [tcp/http] succeeded!\nConnection to 127.0.0.1 8000 port [tcp/*] succeeded!\nConnection to 127.0.0.1 8080 port [tcp/http-alt] succeeded!\nConnection to 127.0.0.1 8443 port [tcp/*] succeeded!\nConnection to 127.0.0.1 9000 port [tcp/*] succeeded!\nConnection to 127.0.0.1 9876 port [tcp/*] succeeded!\n"
                    },
                    {
                        "Start": "2024-07-22T10:32:22.0108352+02:00",
                        "End": "2024-07-22T10:32:22.097720314+02:00",
                        "ExitCode": 0,
                        "Output": "Connection to 127.0.0.1 80 port [tcp/http] succeeded!\nConnection to 127.0.0.1 8000 port [tcp/*] succeeded!\nConnection to 127.0.0.1 8080 port [tcp/http-alt] succeeded!\nConnection to 127.0.0.1 8443 port [tcp/*] succeeded!\nConnection to 127.0.0.1 9000 port [tcp/*] succeeded!\nConnection to 127.0.0.1 9876 port [tcp/*] succeeded!\n"
                    }
                ]
            }
        },
        "Image": "sha256:72adeb05df17e2160b8d766513bdb4d4e1757e52bc2af8894b7e8c2f7cb5310a",
        "ResolvConfPath": "/var/lib/docker/containers/ef49cdd05fcf4b8f1cd7871ac01fe1a7bfd7d154cba56cae57721bcabff0511f/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/ef49cdd05fcf4b8f1cd7871ac01fe1a7bfd7d154cba56cae57721bcabff0511f/hostname",
        "HostsPath": "/var/lib/docker/containers/ef49cdd05fcf4b8f1cd7871ac01fe1a7bfd7d154cba56cae57721bcabff0511f/hosts",
        "LogPath": "/var/lib/docker/containers/ef49cdd05fcf4b8f1cd7871ac01fe1a7bfd7d154cba56cae57721bcabff0511f/ef49cdd05fcf4b8f1cd7871ac01fe1a7bfd7d154cba56cae57721bcabff0511f-json.log",
        "Name": "/nextcloud-aio-mastercontainer",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "docker-default",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": [
                "/var/run/docker.sock:/var/run/docker.sock:ro"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "nextcloud_default",
            "PortBindings": {
                "8080/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "8080"
                    }
                ]
            },
            "RestartPolicy": {
                "Name": "always",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "ConsoleSize": [
                0,
                0
            ],
            "CapAdd": null,
            "CapDrop": null,
            "CgroupnsMode": "private",
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": [],
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": null,
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": null,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "Mounts": [
                {
                    "Type": "volume",
                    "Source": "nextcloud_aio_mastercontainer",
                    "Target": "/mnt/docker-aio-config",
                    "VolumeOptions": {}
                }
            ],
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware",
                "/sys/devices/virtual/powercap"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/bad4d496998032e8c523dca44e6c13886e3851c7ff97862028f841d239378b09-init/diff:/var/lib/docker/overlay2/e9e024cb0f1c02a8e99633c69b5d685abb8f9da0784bce3b86b2355f783ed4d7/diff:/var/lib/docker/overlay2/ecd940dce9caafe5dbf773bc4fb429052766ecfa523c745f32c086c28a602661/diff:/var/lib/docker/overlay2/859db0e23a20728d58a13fe97344bff97fbacc7c69a12f44763aad73a2ef8d2c/diff:/var/lib/docker/overlay2/058ac6ebb1a4b7eb3ccbe9825c5013ee36a7ea291b2e6fc5787962bfe60a95ee/diff:/var/lib/docker/overlay2/f35714f6e8a806d75571e2e328b7157db7551fd3d972bb9cdb725da917350b8e/diff:/var/lib/docker/overlay2/9f658f3d4171690d20c977cedca22b583432d64d2e60c8965cb78d96beb8de4d/diff:/var/lib/docker/overlay2/2d9b0fa4186998b691b636a43a59203ca2475590f8f8c3cb0e7a92b663029e95/diff:/var/lib/docker/overlay2/9f65aa2f0e2f2d70cf04bc9c4df00125c88088c417755464d904123c6362ce6b/diff:/var/lib/docker/overlay2/e163ee1dfe153c02660a59e5f979b265b031a90834e708a05303c60d50a56ea2/diff:/var/lib/docker/overlay2/315a77e773db956bc867a4c9a8441cc20350032e69d188a279cd1d5959d94232/diff:/var/lib/docker/overlay2/7f7e7830bb3fa271ea1359101334c201b6dbb5d161d0fe47c28f674fab0754b8/diff:/var/lib/docker/overlay2/b43ca738353102651f7c7e7edd12419664bb133b4289eea90a3f7b4287056799/diff:/var/lib/docker/overlay2/2eab3b8bc1631ac5f7d5d20e12b66ecaee808a1136bcd641a945152183cc0ad7/diff:/var/lib/docker/overlay2/d2326deef83eed5f765de85103d383cfc25b9c76db0877c0b0369ecf3d83d179/diff:/var/lib/docker/overlay2/fbe2c25642e534edb181da7ed760248b7fe72a0527cf2d2754581d770c87eea6/diff:/var/lib/docker/overlay2/24969be6c4782249ee1e01ce000f569a07715981f9d956c75377fba1972198fb/diff:/var/lib/docker/overlay2/8d647c27601eb7fba7ac9c2d3b1ce95bf0044f66f6f55300940131aa9699ebd6/diff:/var/lib/docker/overlay2/db16a62e7f2211908176591372cb865e093afe8c2e0c50546c6a49e062c932bd/diff",                "MergedDir": "/var/lib/docker/overlay2/bad4d496998032e8c523dca44e6c13886e3851c7ff97862028f841d239378b09/merged",
                "UpperDir": "/var/lib/docker/overlay2/bad4d496998032e8c523dca44e6c13886e3851c7ff97862028f841d239378b09/diff",
                "WorkDir": "/var/lib/docker/overlay2/bad4d496998032e8c523dca44e6c13886e3851c7ff97862028f841d239378b09/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [
            {
                "Type": "volume",
                "Name": "nextcloud_aio_mastercontainer",
                "Source": "/var/lib/docker/volumes/nextcloud_aio_mastercontainer/_data",
                "Destination": "/mnt/docker-aio-config",
                "Driver": "local",
                "Mode": "z",
                "RW": true,
                "Propagation": ""
            },
            {
                "Type": "bind",
                "Source": "/var/run/docker.sock",
                "Destination": "/var/run/docker.sock",
                "Mode": "ro",
                "RW": false,
                "Propagation": "rprivate"
            }
        ],
        "Config": {
            "Hostname": "ef49cdd05fcf",
            "Domainname": "",
            "User": "root",
            "AttachStdin": false,
            "AttachStdout": true,
            "AttachStderr": true,
            "ExposedPorts": {
                "80/tcp": {},
                "8080/tcp": {},
                "8443/tcp": {},
                "9000/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "APACHE_PORT=11000",
                "NEXTCLOUD_DATADIR=/tank/share/ncdata",
                "NEXTCLOUD_MEMORY_LIMIT=2048M",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "PHPIZE_DEPS=autoconf \t\tdpkg-dev dpkg \t\tfile \t\tg++ \t\tgcc \t\tlibc-dev \t\tmake \t\tpkgconf \t\tre2c",
                "PHP_INI_DIR=/usr/local/etc/php",
                "PHP_CFLAGS=-fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64",
                "PHP_CPPFLAGS=-fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64",
                "PHP_LDFLAGS=-Wl,-O1 -pie",
                "GPG_KEYS=1198C0117593497A5EC5C199286AF1F9897469DC C28D937575603EB4ABB725861C0779DC5C0A9DE4 AFD8691FDAEDF03BDF6E460563F15A9B715376CA",
                "PHP_VERSION=8.3.9",
                "PHP_URL=https://www.php.net/distributions/php-8.3.9.tar.xz",
                "PHP_ASC_URL=https://www.php.net/distributions/php-8.3.9.tar.xz.asc",
                "PHP_SHA256=bf4d7b8ea60a356064f88485278bd6f941a230ec16f0fc401574ce1445ad6c77"
            ],
            "Cmd": null,
            "Healthcheck": {
                "Test": [
                    "CMD-SHELL",
                    "/healthcheck.sh"
                ]
            },
            "Image": "nextcloud/all-in-one:latest",
            "Volumes": null,
            "WorkingDir": "/var/www/docker-aio",
            "Entrypoint": [
                "/start.sh"
            ],
            "OnBuild": null,
            "Labels": {
                "com.docker.compose.config-hash": "ad72f8fc7a2c062a6cf0a3a649ed304266e252cfd5585e37158711bb6ef78856",
                "com.docker.compose.container-number": "1",
                "com.docker.compose.depends_on": "",
                "com.docker.compose.image": "sha256:72adeb05df17e2160b8d766513bdb4d4e1757e52bc2af8894b7e8c2f7cb5310a",
                "com.docker.compose.oneoff": "False",
                "com.docker.compose.project": "nextcloud",
                "com.docker.compose.project.config_files": "/nvme/docker/composefiles/nextcloud/nextcloud.yml",
                "com.docker.compose.project.environment_file": "/nvme/docker/composefiles/nextcloud/nextcloud.env,/nvme/docker/composefiles/global.env",
                "com.docker.compose.project.working_dir": "/nvme/docker/composefiles/nextcloud",
                "com.docker.compose.replace": "350ac41e9dcedc9cc4028fb787788e4aca29e45426bf81c8c8aed6b7e677c9d2",
                "com.docker.compose.service": "nextcloud-aio-mastercontainer",
                "com.docker.compose.version": "2.28.1"
            },
            "StopSignal": "SIGQUIT"
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "8d41f157f7c36708880e691f2a51278903971e26140dfc1aa840d470bc5fac28",
            "SandboxKey": "/var/run/docker/netns/8d41f157f7c3",
            "Ports": {
                "80/tcp": null,
                "8080/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "8080"
                    },
                    {
                        "HostIp": "::",
                        "HostPort": "8080"
                    }
                ],
                "8443/tcp": null,
                "9000/tcp": null
            },
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "nextcloud-aio": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "MacAddress": "02:42:ac:13:00:05",
                    "DriverOpts": null,
                    "NetworkID": "6856c189a0d13528c532faa6e19dda1262461096301382d00280948ed45a271d",
                    "EndpointID": "b352f9fcc8a122bbf336a7a1e470ff5d7a035b2d1e6e942122f21b3f50ff7fc7",
                    "Gateway": "172.19.0.1",
                    "IPAddress": "172.19.0.5",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "DNSNames": [
                        "nextcloud-aio-mastercontainer",
                        "ef49cdd05fcf"
                    ]
                },
                "nextcloud_default": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "nextcloud-aio-mastercontainer",
                        "nextcloud-aio-mastercontainer"
                    ],
                    "MacAddress": "02:42:ac:12:00:02",
                    "DriverOpts": null,
                    "NetworkID": "0eafb2a9ba96a7586360c7ed3f8821e69cea8931cc75dc84eff0d10477ecedee",
                    "EndpointID": "475bbe8af975528e5471301b1e7a7f5f2b037b23dede1ea4332e38ca571a24d2",
                    "Gateway": "172.18.0.1",
                    "IPAddress": "172.18.0.2",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "DNSNames": [
                        "nextcloud-aio-mastercontainer",
                        "ef49cdd05fcf"
                    ]
                }
            }
        }
    }
]

**Comment by szaimen** [comment]
Thanks!

Can you post the output of `sudo du -sh /tank/share/ncdata/appdata*/preview` here?

**Reply by daschmidt1994** [reply]
33G     /tank/share/ncdata/appdata_ocg9n1pojooz/preview

**Reply by szaimen** [reply, marked as answer]
This doesnt explain why the backup is now 1tb in size. I rather think it has other reasons. So excluding previews will not solve the problem.

**Reply by daschmidt1994** [reply]
ok my nc data have 208G then I look forward.

But it doesen't make sense to backup the preview or I'm wrong?

**Reply by szaimen** [reply]
The previews are a core-part of Nextcloud and thus need to be included in the backup. You might be able to exclude it by so: https://github.com/nextcloud/all-in-one/discussions/4982#discussioncomment-10075586

However this will break previews upon restore as it will erase them.
